weldx/transformations/cs_manager.py

```diff
@@ -43,14 +43,55 @@ class CoordinateSystemManager:
 
     """
 
-    _id_gen = itertools.count()
-
     @dataclass
-    class CoordinateSystemData:
-        """Class that stores data and the coordinate system, the data is assigned to."""
+    class SubsystemData:
+        """Contains information about subsystems."""
+
+        root: str
+        common_node: str
+        common_node_data: List[str]
+        common_node_neighbors: List[str]
+        time_ref: pd.Timestamp
+        members: List[str]
+        sub_systems: Dict[str, CoordinateSystemManager.SubsystemData]
+        nodes: List[str] = None
+
+        def __eq__(self, other):
+            if not isinstance(other, CoordinateSystemManager.SubsystemData):
+                raise NotImplementedError
+            if self.root != other.root:
+                return False
+            if self.common_node != other.common_node:
+                return False
+            for data in self.common_node_data:
+                if data not in other.common_node_data:
+                    return False
+            for neighbor in self.common_node_neighbors:
+                if neighbor not in other.common_node_neighbors:
+                    return False
+            if self.time_ref != other.time_ref:
+                return False
+            for member in self.members:
+                if member not in other.members:
+                    return False
+            for k, v in self.sub_systems.items():
+                other_v = other.sub_systems.get(k)
+                if other_v is None:
+                    return False
+                if v != other_v:
+                    return False
+            if self.nodes is None:
+                if other.nodes is not None:
+                    return False
+            else:
+                if other.nodes is None:
+                    return False
+                for node in self.nodes:
+                    if node not in other.nodes:
+                        return False
+            return True
```

Comment:
This can be simplified using `compare_nested` as follows:

~~~ python
        def __eq__(self, other):
            if not isinstance(other, CoordinateSystemManager.SubsystemData):
                raise NotImplementedError
            return util.compare_nested(self.__dict__, other.__dict__)
~~~

Problem is that compare nested requires that the order of list items is identical, which is not guaranteed during deserialization. I tried using sets instead, but it seems that `compare_nested` does not support them yet (see #495 )